### PR TITLE
Fix: Parser merge sequence in config (fixes #6158)

### LINKED ISSUE #6158: 2.10 parse error
**What version of ESLint are you using?**

2.10

**What parser (default, Babel-ESLint, etc.) are you using?**

Tried both default and babel-eslint

**Please show your full configuration:**

```
{
	"extends": "ember",

	"parser": "babel-eslint",

	"ecmaFeatures": {
		modules: true,
		blockBindings: true,
		arrowFunctions: true,
		objectLiteralShorthandMethods: true,
		objectLiteralShorthandProperties: true,
		templateStrings: true
	},

	"rules": {
		"quote-props": [2, "consistent-as-needed"],
		"no-ternary": 0,
		"newline-per-chained-call": 0,
		"arrow-body-style": 0,
		"no-warning-comments": 0,
		"max-len": [1, 200],
		"camelcase": 0,
		"space-before-function-paren": [1, "never"],
		"one-var": [ 1, "never" ],
		"indent": [2, 2, {SwitchCase: 1}],
		"object-shorthand": [ 2, "always" ],
		"prefer-const": 0,
		"comma-dangle": 0,
		"spaced-comment": 1,
		"object-curly-spacing": 0,
		"arrow-spacing": [ 1, {
			before: true,
			after: true
		} ],
		"no-restricted-syntax": 0,

		"no-console": 0,
		"no-magic-numbers": 0,
		"valid-jsdoc": [2, {
			"requireReturn": false
		}]
	},

	"globals": {
		"Ember": true,
		"$": true,
		"Edgar": true,
		"visit": true,
		"andThen": true,
		"find": true,
		"findWithAssert": true,
		"click": true,
		"triggerEvent": true,
		"wait": true,
		"ok": true,
		"equal": true,
		"notEqual": true,
		"deepEqual": true,
		"strictEqual": true,
		"throws": true,
		"fillIn": true,
		"exists": true,
		"currentPath": true,
		"currentURL": true,
		"currentRouteName": true,
		"breadcrumbValidator": true,
		"renderProfiler": true,
		"expect": true,
		"_": true,
		"test": true,
		"module": true,
		"YAMP": true,
		"App": true
	}
}

```

**What did you do? Please include the actual source code causing the issue.**

```js
test('Create New Rule', async function() {
});
```

**What did you expect to happen?**

No parse error

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Workspace/tests/acceptance/rule-test.js
  1:31  error  Parsing error: Unexpected token function
```

### LINKED ISSUE #6518: Docs: Confirm block comment approach for style guide
> Note: I'm happy to manually update literally every rule doc over the coming weeks to make the block comments consistent. In this discussion I would suggest prioritising "What is best for the reader" rather than "What is least work for @aubergine10".
## Spacing

It's become clear that we should describe how eslint block comments should be spaced to ensure consistency throughout documentation.

My personal preference is nitrogen-packed as follows:

```
/* eslint rule-name: [ "off", { "prop": val } ] */ 
```

I feel this greatly enhances legibility, compare to the suction-packed version:

```
/*eslint rule-name: ["off", {"prop": val}]*/ 
```

While it takes up less space, it's more difficult to read IMO. In particular, comments don't get syntax highlighting to differentiate brackets, quotes, strings, numbers, etc.
## Quoting

Should the rule-name be "double-qouted"?

My personal preference is that it shouldn't (looks cleaner without quotes), but not doing so breaks consistency compared to the `.eslintrc` and `package.json`.

There is also consistency between the rule name and the options, eg. in the example above `"prop"` is quoted. Arguably, if we're not quoting the rule name, the prop keys shouldn't be quoted either.

There's also the argument that because prop names include non-identifier characters, they should be quoted.

See [quote-props rule](http://eslint.org/docs/rules/quote-props) for potential approaches we could consider.

Discuss :)

### PR DESCRIPTION
**Pull request by gyandeeps**
Correct solution to what was done in #6108

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -243,25 +243,41 @@ Config.prototype.getConfig = function(filePath) {
...
+    // Step 6: Merge in command line rules
+    if (this.options.rules) {
+        debug("Merging command line rules");
+        config = ConfigOps.merge(config, { rules: this.options.rules });
+    }
+
+    // Step 7: Merge in command line globals
+    config = ConfigOps.merge(config, { globals: this.globals });
+
+    // Only override parser is is passed explicitly through command line or if its not
```
Comment (alberto) [inline lib/config.js]: Spelling: `// Only override parser if it is passed explicitly through the command line or if its not`

File: tests/lib/options.js [hunk 2]
```diff
@@ -356,12 +356,6 @@ describe("options", function() {
 
             assert.equal(currentOptions.parser, "test");
         });
-
-        it("should return a espree if --parser is not passed", function() {
```
Comment (mysticatea) [inline tests/lib/options.js]: Is this no problem?
We have an API to get parser's module: http://eslint.org/docs/developer-guide/working-with-rules#the-context-object

Comment (gyandeeps) [inline tests/lib/options.js]: Sorry, I dint understand what you are trying to say here?
We always default to `espree` if the parser is not defined.. https://github.com/eslint/eslint/blob/master/lib/eslint.js#L435

Comment (mysticatea) [inline tests/lib/options.js]: Thank you for the pointer. I got it.

**Comment by ilyavolodin** [comment]
@gyandeeps I'll merge it to release a patch for bable-eslint users, but could we add some unittests for this as a separate chore? Config merging should be pretty unittestable.

**Pull request merged**